lib/retirement_swap/processor.rb

```diff
@@ -0,0 +1,30 @@
+module RetirementSwap
+  class Processor
+    class NullAlgorithm
+      def process(*args)
+        return []
+      end
+    end
+
+    ALGORITHMS = {
+      'swap' => RetirementSwap::SwapAlgorithm
+    }
+
+    attr_reader :workflows
+
+    def initialize(storage, output, config)
+      @storage = storage
+      @output  = output
+      @workflows = config.each.with_object(Hash.new(NullAlgorithm.new)) do |(workflow_id, workflow_config), hash|
+        hash[workflow_id] = ALGORITHMS.fetch(workflow_config.fetch('algorithm')).new(storage, output)
```

Comment:
Line is too long. [101/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.